lib/models/event_schedule.rb

```diff
@@ -0,0 +1,15 @@
+class EventSchedule
+  def self.populate_upcoming_show_and_tell!
+    upcoming_dates = (Time.zone.today..Time.zone.today + 4.weeks).to_a
```

Comment:
You could reduce these 2 lines to:
`(Time.zone.today..Time.zone.today + 4.weeks).select(&:friday?)`